packages/client/src/schema/repository.ts

```diff
@@ -2013,6 +3349,93 @@ export class RestRepository<Record extends XataRecord>
   }
 }
 
+export const initObjectKysely = <T>(
+  repo: KyselyRepository<any>,
+  schemaTables: Schemas.Table[],
+  table: string,
+  object: Record<string, any>,
+  selectedColumns: SelectableColumn<T>[] | SelectableColumnWithObjectNotation<T>[]
+) => {
+  const data: Dictionary<unknown> = {};
+  Object.assign(data, { ...object });
+
+  const { columns } = schemaTables.find(({ name }) => name === table) ?? {};
+  if (!columns) console.error(`Table ${table} not found in schema`);
+
+  for (const column of columns ?? []) {
+    // Ignore columns not selected
+    if (!isValidColumn(selectedColumns, column)) continue;
+
+    const value = data[column.name];
+
+    switch (column.type) {
+      case 'datetime': {
+        const date = value !== undefined && value !== null ? new Date(value as string) : null;
+
+        if (date !== null && isNaN(date.getTime())) {
+          console.error(`Failed to parse date ${value} for field ${column.name}`);
+        } else {
+          data[column.name] = date;
+        }
+
+        break;
+      }
+      case 'file':
+        data[column.name] = isDefined(value) ? new XataFile(value as any) : null;
+        break;
+      case 'file[]':
+        data[column.name] = (value as XataArrayFile[])?.map((item) => new XataFile(item)) ?? null;
+        break;
+      case 'json':
+        data[column.name] = parseJson(value as string);
+        break;
+      default:
+        data[column.name] = value ?? null;
+
+        if (column.notNull === true && value === null) {
+          console.error(`Parse error, column ${column.name} is non nullable and value resolves null`);
+        }
+        break;
+    }
+  }
+
+  const record = { ...data };
+
+  record.read = async function (columns?: any) {
+    return repo.read(record['xata_id'] as string, columns);
+  };
+
+  record.update = async function (data: any, b?: any) {
+    const columns = isValidSelectableColumns(b) ? b : ['*'];
+    return repo.update(record['xata_id'] as string, data, columns);
+  };
+
+  record.replace = async function (data: any, b?: any) {
+    const validColumns = isValidSelectableColumns(b) ? b : ['*'];
+    return repo.createOrReplace(record['xata_id'] as string, data, validColumns);
+  };
+
+  record.delete = async function () {
+    return repo.delete(record['xata_id'] as string);
```

Comment:
I've created a new private field called `#primaryKey` on KyselyRepository and it tries to get the primaryKey off schema. I used the `BaseData` type for now instead of `Identifiable` so that it compiles but yeah I'll need help fixing the types to be correct :D